File: 153-find-minimum-in-rotated-sorted-array/153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int left = 0;
        int right = nums.length-1;
        if(nums[left]<=nums[right]){
            return nums[left];
        }
        
        while(left<right){
            int mid = (left+right)/2;
            if(nums[left]<nums[mid]){
                left=mid;
            }
            else{
                right=mid;
            }
        }
        return nums[left+1];
    }
}

File: 153-find-minimum-in-rotated-sorted-array/153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums)-1
        if nums[left] <= nums[right]:
            return nums[left]
        
        while(left<right):
            mid = (left+right)//2
            if nums[left]<nums[mid]:
                left=mid
            else:
                right=mid
            
        return nums[left+1]